### PR TITLE
add isobaric correction defaults for tmt6plex and 11plex

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description
correction matrix
- [x] add tmt6plex
- [x] add tmt11plex

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
bug_fix

___

### **Description**
- Removed the special case handling for "tmt11plex" in the `IsobaricIsotopeCorrector` where the correction matrix being an identity matrix was previously allowed.
- Now, an exception is consistently thrown if the correction matrix is an identity matrix, ensuring that a valid isotope correction matrix is provided.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>IsobaricIsotopeCorrector.cpp</strong><dd><code>Remove special case for "tmt11plex" in correction matrix validation</code></dd></summary>
<hr>

src/openms/source/ANALYSIS/QUANTITATION/IsobaricIsotopeCorrector.cpp

<li>Removed special case handling for "tmt11plex" when the correction <br>matrix is the identity matrix.<br> <li> Ensured that an exception is thrown for identity matrices, enforcing <br>the need for a valid correction matrix.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7601/files#diff-d3bf0f78ef87c457cd8cd4c084a956316bda48351fed5aeffe114647988ad3a2">+3/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: src/openms/source/ANALYSIS/QUANTITATION/IsobaricIsotopeCorrector.cpp [hunk 1]
```diff
@@ -43,14 +43,10 @@ namespace OpenMS
     {
       OPENMS_LOG_DEBUG << "Correction matrix is the identity matrix." << std::endl;
       OPENMS_LOG_DEBUG << correction_matrix << std::endl;
-
-      // workaround: TMT11plex has a special case where the correction matrix is the identity matrix
-      if (quant_method->getMethodName() != "tmt11plex")
-      {        
-        throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+      
+      throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/QUANTITATION/IsobaricIsotopeCorrector.cpp:47]: special treatment of tmt11plex (and 6plex) not needed anymore

File: src/tests/class_tests/openms/source/TMTSixPlexQuantitationMethod_test.cpp [hunk 2]
```diff
@@ -113,10 +113,8 @@ START_SECTION((virtual Matrix<double> getIsotopeCorrectionMatrix() const ))
   {
     for(size_t j = 0; j < m.cols(); ++j)
     {
-      if (i == j)
-        TEST_REAL_SIMILAR(m(i,j), 1.0)
-      else
-        TEST_REAL_SIMILAR(m(i,j), 0.0)
+      if (i == j) { TEST_EQUAL(m(i,j) > 0.5 } // diagonal entries should be largest
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/source/TMTSixPlexQuantitationMethod_test.cpp]:
```suggestion
      if (i == j) { TEST_EQUAL(m(i,j) > 0.5) } // diagonal entries should be largest
```

File: src/tests/class_tests/openms/source/TMTElevenPlexQuantitationMethod_test.cpp [hunk 3]
```diff
@@ -181,8 +181,8 @@ START_SECTION((virtual Matrix<double> getIsotopeCorrectionMatrix() const ))
   {
     for (size_t j = 0; j < m.cols(); ++j)
     {
-      if (i == j) { TEST_REAL_SIMILAR(m(i,j), 1.0) }
-      else { TEST_REAL_SIMILAR(m(i,j), 0.0) }
+      if (i == j) { TEST_EQUAL(m(i,j) > 0.5 } // diagonal entries should be largest
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/source/TMTElevenPlexQuantitationMethod_test.cpp]:
```suggestion
      if (i == j) { TEST_EQUAL(m(i,j) > 0.5 ) } // diagonal entries should be largest
```

**Comment by timosachsenberg** [comment]
this is a hot fix that should go into 3.2.1

**Review by poshul — APPROVED** [review]

**Pull request merged**